pyproject.toml

```diff
@@ -201,35 +201,35 @@ max-line-length = 105  # default 100
 [tool.pylint."messages control"]
 disable = [
 # intentionally disabled:
-    "spelling",  # too noisy
+    "arguments-renamed", # mostly used in the transpiler, renaming seems natural
+    "docstring-first-line-empty", # relax docstring style
+    "duplicate-code", # disabled as it is too verbose
     "fixme", # disabled as TODOs would show up as warnings
+    "import-outside-toplevel", "import-error", # overzealous with our optionals/dynamic packages
+    "nested-min-max", # this gives false equivalencies if implemented for the current lint version
+    "no-else-return",  # relax "elif" after a clause with a return
+    "no-member", # for dynamically created members
+    "no-value-for-parameter", # cannot implement for eldritch horrors: qiskit/providers/options.py
+    "not-context-manager", # no need for QuantumCircuit to implement __enter__ and __exit__
```

Comment:
I think `not-context-manager` is actually indicative of some error in the type hints around `QuantumCircuit.if_test`, `while_loop`, `switch` and `for_loop`, but I'm not certain.  We might want to leave that one in the "needs sorting out" list.

Fwiw, I think the problem is maybe that the "context manager" overload is specified with the wrong signature - I think the overload for `if_test` should (for example) be `def if_test(self, condition: tuple[ClassicalRegister | Clbit, int]) -> IfContext: ...` rather than what it is at the moment where it falsely includes all the other arguments as well.

Reply:
moved this back to temporary. will probably look at it again soon